File: wikz_web/models/Publicacion.php
<?php

// models/Publicacion.php
require_once 'config/Database.php';

class Publicacion
{
    private $pdo;

    public $id;
    public $id_usuario;
    public $titulo;
    public $descripcion;
    public $imagen;
    public $fecha_creacion;

    public function __construct()
    {
        $this->pdo = Database::getConnection();
    }

    // Crear publicación
    public function crear($id_usuario, $titulo, $descripcion, $imagen = null)
    {
        $stmt = $this->pdo->prepare(
            "INSERT INTO publicaciones (id_usuario, titulo, descripcion, imagen, creacion) VALUES (?, ?, ?, ?, NOW())"
        );
        return $stmt->execute([$id_usuario, $titulo, $descripcion, $imagen]);
    }

    // Obtener publicaciones de un usuario
    public function getPorUsuario($id_usuario)
    {
        $stmt = $this->pdo->prepare("SELECT * FROM publicaciones WHERE id_usuario = ? ORDER BY creacion DESC");
        $stmt->execute([$id_usuario]);
        return $stmt->fetchAll(PDO::FETCH_ASSOC);
    }

    // Obtener publicación por id
    public function getById($id)
    {
        $stmt = $this->pdo->prepare("SELECT * FROM publicaciones WHERE id = ?");
        $stmt->execute([$id]);
        return $stmt->fetch(PDO::FETCH_ASSOC);
    }
}

File: wikz_escritorio/wikz_escritorio/Modelos/Publicacion.cs
using System;
using Newtonsoft.Json;

namespace wikz_escritorio.Modelos
{
    public class Publicacion
    {
        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("idUsuario")]
        public int IdUsuario { get; set; }

        [JsonProperty("titulo")]
        public string Titulo { get; set; }

        [JsonProperty("descripcion")]
        public string Descripcion { get; set; }

        [JsonProperty("imagenBase64")]
        public string ImagenBase64 { get; set; }

        [JsonProperty("fechaCreacion")]
        public string FechaCreacion { get; set; }

        public Publicacion()
        {
            Titulo = "";
            Descripcion = "";
            FechaCreacion = DateTime.Now.ToString();
        }

        public override string ToString()
        {
            return $"{Titulo} (ID: {Id})";
        }
    }
}